opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -349,8 +349,10 @@ static int accelerator_cuda_memcpy_async(int dest_dev_id, int src_dev_id, void *
     CUresult result;
 
     int delayed_init = opal_accelerator_cuda_delayed_init();
-    if (OPAL_UNLIKELY(0 != delayed_init)) {
-        return delayed_init;
+    if (0 != delayed_init) {
+        /* If initialization failed, it's likely this may be host buffers */
+        memcpy(dest, src, size);
+        return OPAL_SUCCESS;
```

Comment:
I am trying to understand  the significance of this change.  I am fine and understand with the fact that the component returns 'host memory' for all memory types, but what is the scenario in which case the async_mempcy function (or any of the other functions below) of the accelerator framework are being called for host memory?

Reply:
I originally added it because this path was being entered in opal_datatype_accelerator_memcpy. I should do more of a deep dive and see why it doesn't use regular memcpy.

Reply:
Turns out there were checks missing from the opal_datatype_copy functions. I suppose there was no issue prior because the accelerator memcpy could handle host - host as long as the cuda was initialized.